ui/src/ducks/app/solutions.js

```diff
@@ -217,26 +230,42 @@ export function* updateEnvironments(environments) {
           const availableDowngradeVersion = [];
 
           availableSolutionVersions.versions.forEach(avaSolVer => {
-            if (avaSolVer.version.localeCompare(currentSolutionVersion) <= -1) {
+            if (
+              avaSolVer.version.localeCompare(
+                isDeploySolutionSucceed
+                  ? currentSolutionVersion
+                  : solutionOperatorDeploymentVersion,
```

Comment:
No need for this check, both branches always yield the `solutionOperatorDeploymentVersion` anyway.

What would like to show differently if the configured version is different from the deployed one? For me, the only  thing to show would be that I can only see a "apply" button if there's a difference between `current` and `observed` versions.

Reply:
Yes, you are right! We should go for Operator Version anyways because it's the actual deployed version.